.gitlab-ci.yml

```diff
@@ -29,27 +30,67 @@ test:
     paths:
       - unit-coverage.tar
 
-compile:
-  stage: build
-  script:
-    - make build V=1
-    - cp mender $CI_PROJECT_DIR
-  artifacts:
-    expire_in: 2w
-    paths:
-      - mender
-
 publish:tests:
+  image: alpine
   stage: publish
+  before_script:
+    - apk add --no-cache bash curl findutils git
   dependencies:
     - test
   script:
     - tar -xvf unit-coverage.tar
-    # - bash <(curl -s https://codecov.io/bash) -s ./tests/unit-coverage;
+    - bash -c "bash <(curl -s https://codecov.io/bash) -Z -s ./tests/unit-coverage"
 
-publish:binary:
-  stage: publish
-  dependencies:
-    - compile
+trigger:mender-qa:
+  image: alpine
+  stage: trigger
+  before_script:
+    - apk add --no-cache git curl python3
+    - pip3 install pyyaml
+    - git clone https://github.com/mendersoftware/integration.git integration
+    - cd integration
   script:
-    - echo "publish mender binary here"
\ No newline at end of file
+    # The embbedded script below will
+    # for each integration version including mender version $CI_COMMIT_REF_NAME:
+    # * construct a curl formatted string concatenating each repo and revision like:
+    #   '-F variables[repoA_REV]=X.X.x -F variables[repoB_REV]=X.X.x ...'
+    # * trigger a Mender QA pipeline to build/test/publish client
+    - for integ_version in $(./extra/release_tool.py
+                             --integration-versions-including mender
+                             --version 2.1.x); do
```

Comment:
Hardcoded `2.1.x`?

Reply:
Ouch! That was while debugging...

Reply:
It should be `$CI_COMMIT_REF_NAME`. Fixed now